src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -785,8 +785,10 @@ abstract class UnCurry extends InfoTransform
       if (dd.symbol.isConstructor)
         reporter.error(dd.symbol.pos, "A constructor cannot be annotated with a `varargs` annotation.")
       else {
-        val hasRepeated = mexists(dd.symbol.paramss)(sym => definitions.isRepeatedParamType(sym.tpe))
-        if (!hasRepeated) reporter.error(dd.symbol.pos, "A method without repeated parameters cannot be annotated with the `varargs` annotation.")
+        val initPs :+ lastPs = dd.symbol.paramss
```

Comment:
```suggestion
        val initPs :+ lastPs = dd.symbol.paramss.filter(_.nonEmpty)
```

Reply:
Wait, I guess this could fail when you only have empty param lists. But I guess the pre-suggestion code was already prone to failure when you no param list.